src/RazorSlices/RazorSlice.cs

```diff
@@ -493,6 +507,12 @@ private void WriteValue<T>(T value)
             Write(((byte[])(object)value).AsSpan());
         }
         // Handle derived types (this currently results in value types being boxed)
+#if NET8_0_OR_GREATER
+        else if (value is IUtf8SpanFormattable)
+        {
+            WriteUtf8SpanFormattable((IUtf8SpanFormattable)(object)value, default, null);
```

Comment:
nit:
```
        else if (value is IUtf8SpanFormattable formattable)
        {
            WriteUtf8SpanFormattable(formattable, default, null);
```
Saves an unnecessary `castclass` call in IL, although JIT might be smart enough to remove it?

Reply:
I was told this is the pattern to use as JIT supports it and will avoid the boxing, etc.
Cc @StephenToub

Reply:
Huh, that's interesting. Any reason the JIT can't produce the same code in both cases?